sdks/python/apache_beam/io/iobase.py

```diff
@@ -737,7 +740,27 @@ def open_writer(self, init_result, uid):
     """
     raise NotImplementedError
 
-  def finalize_write(self, init_result, writer_results):
+  def pre_finalize(self, init_result, writer_results):
```

Comment:
<!--new_thread; commit:bc394b7dd981af5640e2e80ebcc83e86eee0136e; resolved:0-->
Can we do this using a ParDo/DoFn instead of updating the sink interface ? Sink interface has been removed from Java and Python should follow in the near future.

Reply:
As discussed elsewhere, writer_results are a Sink implementation detail and thus we can't implement a generic pre_finalize ParDo here.